lib/Doctrine/ORM/EntityManager.php

```diff
@@ -813,44 +816,6 @@ public function initializeObject($obj)
     }
 
     /**
-     * Factory method to create EntityManager instances.
-     *
-     * @param mixed         $conn         An array with the connection parameters or an existing Connection instance.
-     * @param Configuration $config       The Configuration instance to use.
-     * @param EventManager  $eventManager The EventManager instance to use.
-     *
-     * @return EntityManager The created EntityManager.
-     *
-     * @throws \InvalidArgumentException
-     * @throws ORMException
-     */
-    public static function create($conn, Configuration $config, EventManager $eventManager = null)
```

Comment:
As for now EntityManager has a lot of responsibilities including creation of its own instance. Separate class should prepare and instantiate EntityManager. I agree that this is backward incompatible change. The strategy might be changed based on release policy. In case 2.5.* allows to introduce this change, we are in a good shape. Alternatively, this method might marked as @deprecated since 2.5 and proxy behavior to the EntityManagerFactory
Like this:
{code}
    /**
     * @deprecated since 2.5
     */
    public static function create($conn, Configuration $config, EventManager $eventManager = null)
    {
        return new EntityManagerFactory($conn, $config, $eventManager);
    }
{code}

Thanks